lib/auth/auth_with_roles.go

```diff
@@ -6752,17 +6769,64 @@ func verbsToReplaceResourceWithOrigin(stored types.ResourceWithOrigin) []string
 	return verbs
 }
 
-// checkOktaAccess gates access to update operations on user records based
+// checkOktaUserAccess gates access to update operations on user records based
 // on the origin label on the supplied user record.
 //
-// # See usersv1.CheckOktaAccess() for the actual access rules
+// # See okta.CheckAccess() for the actual access rules
 //
 // TODO(tcsc): Delete in 16.0.0 when user management is removed from `ServerWithRoles`
-func checkOktaAccess(ctx context.Context, authzCtx *authz.Context, users services.UsersService, existingUsername string, verb string) error {
+func checkOktaUserAccess(ctx context.Context, authzCtx *authz.Context, users services.UsersService, existingUsername string, verb string) error {
 	existingUser, err := users.GetUser(ctx, existingUsername, false)
 	if err != nil && !trace.IsNotFound(err) {
 		return trace.Wrap(err)
 	}
 
-	return usersv1.CheckOktaAccess(authzCtx, existingUser, verb)
+	return okta.CheckAccess(authzCtx, existingUser, verb)
+}
+
+// checkOktaLockTarget prevents the okta service from locking users that are not
+// controlled by the Okta service.
+func checkOktaLockTarget(ctx context.Context, authzCtx *authz.Context, users services.UserGetter, lock types.Lock) error {
+	const errorMsg = "Okta service may only lock okta user"
+
+	if !authz.HasBuiltinRole(*authzCtx, string(types.RoleOkta)) {
+		return nil
+	}
+
+	target := lock.Target()
+	switch {
+	case target.AccessRequest != "", target.Device != "", target.Login != "",
+		target.MFADevice != "", target.Node != "", target.Role != "",
+		target.ServerID != "", target.WindowsDesktop != "":
+		return trace.BadParameter(errorMsg)
+
+	case target.User == "":
+		return trace.BadParameter(errorMsg)
+	}
```

Comment:
```suggestion
	switch {
	case !proto.Equal(&target, &types.LockTarget{
		User: target.User,
	}):
		return trace.BadParameter(errorMsg)

	case target.User == "":
		return trace.BadParameter(errorMsg)
	}
```

This seems to be a better check if we add other lock targets